force-app/main/default/classes/LeadTriggerTest.cls

```diff
@@ -0,0 +1,312 @@
+        List<Lead> updatedLeads = new List<Lead>();
+
+        List<Lead> ldListCheckVP = [select Id, Title from Lead where Id = :leadVP.Id];
+        List<Lead> ldListCheckMgr = [select Id, Title from Lead where Id = :leadMgr.Id];
+        List<Lead> ldListCheckExec = [select Id, Title from Lead where Id = :leadExec.Id];
+        List<Lead> ldListCheckAssist = [select Id, Title from Lead where Id = :leadAssist.Id];
+        
+        ldListCheckVP[0].Title = 'vp';
+        ldListCheckMgr[0].Title = 'mgr';
+        ldListCheckExec[0].Title = 'exec';
+        ldListCheckAssist[0].Title = 'assist';
+       
+        updatedLeads.add(ldListCheckVP[0]);
+        updatedLeads.add(ldListCheckMgr[0]);
+        updatedLeads.add(ldListCheckExec[0]);
+        updatedLeads.add(ldListCheckAssist[0]);
+
+        // perform test
+        Test.startTest();
+        update updatedLeads;
+        Test.stopTest();
+
+        // grab values to check
+        List<Lead> ldListCheckVPUpdated = [select Id, Title from Lead where Id = :leadVP.Id];
+        List<Lead> ldListCheckMgrUpdated = [select Id, Title from Lead where Id = :leadMgr.Id];
+        List<Lead> ldListCheckExecUpdated = [select Id, Title from Lead where Id = :leadExec.Id];
+        List<Lead> ldListCheckAssistUpdated = [select Id, Title from Lead where Id = :leadAssist.Id];
+
+        // check results
+        System.assertEquals('Vice President', ldListCheckVPUpdated[0].Title, 'VP Title is not Vice President');
+        System.assertEquals('Manager', ldListCheckMgrUpdated[0].Title, 'Mgr Title is not Manager');
+        System.assertEquals('Executive', ldListCheckExecUpdated[0].Title, 'Exec Title is not Executive');
+        System.assertEquals('Assistant', ldListCheckAssistUpdated[0].Title, 'Assist Title is not Assistant');
+    }
+
+    /*
+    * Test inserting various fields to increment scores.
+    */
+    @IsTest
+    static void testHandleAutoLeadScoring_Insert() {
+
+        // create 3 differing Leads to test all if statements/scores
+        List<Lead> ldList = new List<Lead>();
+
+        Lead ldA = new Lead(
+            FirstName='Test', 
+            LastName='Lead', 
+            Company='Test Company', 
+            LeadSource='Website', 
+            Email='test@gmail.com', 
+            Phone='1234567890'); // 8        
+        Lead ldB = new Lead(
+            FirstName='Test', 
+            LastName='Lead', 
+            Company='Test Company', 
+            Industry='Technology', 
+            Phone='123-456-7890'); // 15
+        Lead ldC = new Lead(
+            FirstName='Test', 
+            LastName='Lead', 
+            Company='Test Company', 
+            LeadSource='Website', 
+            Email='test@gmail.com', 
+            Phone='1234567890', 
+            Industry='Technology'); // 18
+        Lead ldD = new Lead(FirstName='Test', LastName='Lead', Company='Test Company'); // 0
+
+        ldList.add(ldA);
```

Comment:
You can condense this to a single line by initializing the list with the Lead records. 
```
        List<Lead> ldList = new List<Lead>{ldA, ldB, ldC, ldD};
```